File: Data Structures & Algorithms/last-stone-weight/submission-1.py
class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:
        if len(stones) == 1:
            return stones[0]
        stones = [-s for s in stones]
        heapq.heapify(stones)

        while len(stones) > 1:
            first = heapq.heappop(stones)
            second = heapq.heappop(stones)
            x = abs(first - second)
            heapq.heappush(stones, -x)

        return abs(stones[0])

File: Data Structures & Algorithms/climbing-stairs/submission-1.cpp
class Solution {
public:
    int climbStairs(int n) {
        int a = 1, b = 1;
        if(n <= 1) return 1;
        int c = 0;
        for(int i = 2; i <= n; i++){
            c = a + b;
            b = a;
            a = c;
        }
        return c;
    }
};
